analysis/src/Hover.ml

```diff
@@ -125,31 +125,44 @@ let getHoverViaCompletions ~debug ~path ~pos ~currentFile ~forHover
     ~supportsMarkdownLinks =
   match Completions.getCompletions ~debug ~path ~pos ~currentFile ~forHover with
   | None -> None
-  | Some (completions, {file; package}) -> (
+  | Some (completions, ({file; package} as full), scope) -> (
+    let rawOpens = Scope.getRawOpens scope in
+    let allFiles =
+      FileSet.union package.projectFiles package.dependenciesFiles
+    in
     match completions with
     | {kind = Label typString; docstring} :: _ ->
       let parts =
         (if typString = "" then [] else [Markdown.codeBlock typString])
         @ docstring
       in
       Some (Protocol.stringifyHover (String.concat "\n\n" parts))
-    | {kind = Field _; docstring} :: _ -> (
-      match CompletionBackEnd.completionsGetTypeEnv completions with
+    | {kind = Field _; env; docstring} :: _ -> (
+      let opens = CompletionBackEnd.getOpens ~debug ~rawOpens ~package ~env in
+      match
+        CompletionBackEnd.completionsGetTypeEnv2 ~debug ~full ~rawOpens ~opens
```

Comment:
I had not noticed much the business of `completionsGetTypeEnv` vs `completionsGetTypeEnv2`. Is there ever a reason for using the former given the latter is more powerful?

Reply:
Probably not. Mostly a bridge for migrating to the new one. Will take care of this as we do our major refactor of all of this new functionality.